### PR TITLE
Add feature test for copy button

### PR DESCRIPTION
**Pull request by leoj3n**
Closes #5954

### CODE REVIEW

File: ui-v2/tests/acceptance/components/copy-button.feature [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    Given 1 datacenter model with the value "dc-1"
+  Scenario: Clicking the copy button
+    When I visit the node page for yaml
+    ---
+      dc: dc-1
+      node: node-0
+      ---
+    Then the url should be /dc-1/nodes/node-0
+    When I click ".healthcheck-output:nth-child(1) button.copy-btn"
+    Then I see the text "Sorry, something went wrong!" in ".healthcheck-output:nth-child(1) p.feedback-dialog-out"
```
Comment (leoj3n) [inline ui-v2/tests/acceptance/components/copy-button.feature]: This is what the button actually says when I run the test suite with:

```console
$ yarn run test -f 'components / copy-button'
```

As well as when I run it normally (`make test`).

File: ui-v2/tests/acceptance/components/copy-button.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    Then I see the text "Copied output!" in ".healthcheck-output:nth-child(1) p.feedback-dialog-out"
```
Comment (leoj3n) [inline ui-v2/tests/acceptance/components/copy-button.feature]: This is what I originally expected the feedback dialog of the copy button to say, as it's what that value is when running the demo frontend with `make start-api` and `make start`.

**Comment by leoj3n** [comment]
```console
$ yarn run test -f 'components / copy-button'
yarn run v1.16.0
$ ember test --test-port=${EMBER_TEST_PORT:-7357} -f 'components / copy-button'
WARNING: Node v10.14.2 is not tested against Ember CLI on your platform. We recommend that you use the most-recent "Active LTS" version of Node.js. See https://git.io/v7S5n for details.
Could not start watchman
Visit https://ember-cli.com/user-guide/#watchman for more info.

/Users/leoj/.go/src/github.com/hashicorp/consul/ui-v2/app/components/confirmation-dialog.js
  12:3  warning  Use closure actions, unless you need bubbling  ember/closure-actions

✖ 1 problem (0 errors, 1 warning)

cleaning up...
Built project successfully. Stored in "/Users/leoj/.go/src/github.com/hashicorp/consul/ui-v2/tmp/class-tests_dist-ZdhACCWg.tmp".
not ok 1 Chrome 74.0 - Feature: components / copy-button > Scenario: Clicking the copy button: Scenario: Clicking the copy button
    ---
        actual: >
            false
        expected: >
            true
        stack: >
                at Object.<anonymous> (http://localhost:7357/ui/assets/tests.js:9718:14)
                at Object.invoke (http://localhost:7357/ui/assets/test-support.js:4013:19)
                at http://localhost:7357/ui/assets/test-support.js:3653:29
                at http://localhost:7357/ui/assets/test-support.js:3098:13
                at http://localhost:7357/ui/assets/test-support.js:3073:53
                at http://localhost:7357/ui/assets/test-support.js:3094:17
        message: >
            Expected to see "Copied output!" in ".healthcheck-output:nth-child(1) p.feedback-dialog-out"
        Log: |
            { type: 'log', text: '\'GET: /v1/catalog/datacenters\'\n' }
            { type: 'log', text: '\'GET: /v1/coordinate/nodes\'\n' }
            { type: 'log', text: '\'GET: /v1/internal/ui/node/node-0\'\n' }
            { type: 'log', text: '\'GET: /v1/session/node/node-0\'\n' }
    ...

1..1
# tests 1
# pass  0
# skip  0
# fail  1
Testem finished with non-zero exit code. Tests failed.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by johncowen** [comment]
Hi @leoj3n 

I've submitted a PR related to this, there is some more info there but the gist of it is, it doesn't look like we can interact with the clipboard in an automated manner (which kinda makes sense)

https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand

> Note: Only returns true if part of a user interaction. Don't try using the return value to verify browser support before calling a command.

If you try to interact with the clipboard as a result of something else (not a user interaction, but say an automated test), then `execCommand` returns `false`, which in the code we use eventually results in an `error` event. Hence you see the error text in the automated testing.

The above PR should provide us with a way to 'almost' test this end to end with a different type of clipboard, and it's similar to how we test our (almost) end to end HTTP API interaction.

Once the above PR has been merged you should be able to rebase and 🤞 your test should work as expected. I'll keep an eye on the PR and ping you when it's gone in.

Again, if you are too busy or anything, feel free to let me know and I can continue on instead, I'm guessing you are ok to do this though.

Thanks,

John

**Comment by johncowen** [comment]
Hey @leoj3n 

If you pull and rebase `master` you should (🤞) be able to test against the text in that button now. See the above PR for any details. If you have any issues let me know.

Thanks for this again, I love these little things nudge us to build in larger things that will be more more useful for the future 🙌 . We now have a way of testing/asserting all of our clipboard copying functionality 🎉 

Thanks,

John

**Comment by johncowen** [comment]
Hey @leoj3n ,

Thanks again for your work here, appreciate you have other things on, we would like this to be in our next release, so I'm going to merge this soon if I don't hear back from you - I can amend the tests you wrote now I've fixed it so things pass your tests.

👍 

John

**Comment by leoj3n** [comment]
@johncowen Cool, now the inverse test failing:

```
❯ yarn run test -f 'components / copy-button'
yarn run v1.16.0
$ ember test --test-port=${EMBER_TEST_PORT:-7357} -f 'components / copy-button'
WARNING: Node v10.14.2 is not tested against Ember CLI on your platform. We recommend that you use the most-recent "Active LTS" version of Node.js. See https://git.io/v7S5n for details.
Could not start watchman
Visit https://ember-cli.com/user-guide/#watchman for more info.
⠦ BuildingBrowserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`

/Users/leoj/src/oss/consul-1/ui-v2/app/components/confirmation-dialog.js
  12:3  warning  Use closure actions, unless you need bubbling  ember/closure-actions

✖ 1 problem (0 errors, 1 warning)

Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
cleaning up...
Built project successfully. Stored in "/Users/leoj/src/oss/consul-1/ui-v2/tmp/class-tests_dist-keTOvU6T.tmp".
not ok 1 Chrome 75.0 - Feature: components / copy-button > Scenario: Clicking the copy button: Scenario: Clicking the copy button
    ---
        actual: >
            false
        expected: >
            true
        stack: >
                at Object.<anonymous> (http://localhost:7357/ui/assets/tests.js:9624:14)
                at Object.invoke (http://localhost:7357/ui/assets/test-support.js:4013:19)
                at http://localhost:7357/ui/assets/test-support.js:3653:29
                at http://localhost:7357/ui/assets/test-support.js:3098:13
                at http://localhost:7357/ui/assets/test-support.js:3073:53
                at http://localhost:7357/ui/assets/test-support.js:3094:17
        message: >
            Expected to see "Sorry, something went wrong!" in ".healthcheck-output:nth-child(1) p.feedback-dialog-out"
        Log: |
    ...

1..1
# tests 1
# pass  0
# skip  0
# fail  1
Testem finished with non-zero exit code. Tests failed.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I'll remove that test and you should be able to merge with master.

**Comment by leoj3n** [comment]
All UI tests passing on my end:

```console
❯ yarn run test -f 'components / copy-button'
yarn run v1.16.0
$ ember test --test-port=${EMBER_TEST_PORT:-7357} -f 'components / copy-button'
WARNING: Node v10.14.2 is not tested against Ember CLI on your platform. We recommend that you use the most-recent "Active LTS" version of Node.js. See https://git.io/v7S5n for details.
Could not start watchman
Visit https://ember-cli.com/user-guide/#watchman for more info.

/Users/leoj/src/oss/consul-1/ui-v2/app/components/confirmation-dialog.js
  12:3  warning  Use closure actions, unless you need bubbling  ember/closure-actions

✖ 1 problem (0 errors, 1 warning)

cleaning up...
Built project successfully. Stored in "/Users/leoj/src/oss/consul-1/ui-v2/tmp/class-tests_dist-AcJjfU9q.tmp".
ok 1 Chrome 75.0 - Feature: components / copy-button > Scenario: Clicking the copy button: Scenario: Clicking the copy button

1..1
# tests 1
# pass  1
# skip  0
# fail  0

# ok
✨  Done in 33.28s.
```

**Pull request merged**